insights/parsers/softnet_stat.py

```diff
@@ -0,0 +1,100 @@
+per cpu(in row), number of packets processed i.e packet_process (first column),
+number of packet drops packet_drops (second column), time squeeze i.e net_rx_action
+performed time_squeeze(third column),cpu collision i.e collision occur while obtaining
+device lock while transmitting cpu_collision packets (Ninth column), received_rps
+number of times cpu woken up received_rps (tenth column), number of times reached flow
+limit count flow_limit_count (eleventh column).
+
+    The typical contents of ``cat /proc/net/softnet_stat`` file is as below::
+
+        00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+
+    Examples:
+        >>> SOFTNET_STATS = '''
+        ... 00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ...'''.strip()
+        >>> softnet_obj = shared[SoftNetStats]
+        >>> softnet_obj.is_packet_drops()
+        True
+        >>> softnet_obj.get_cpu_instaces()
+        4
+        >>> softnet_obj.per_cpu_nstat('packet_drops')
+        {'cpu2': '00000000', 'cpu3': '00000001', 'cpu0': '00000000', 'cpu1': '00000000'}
+"""
+
+from .. import Parser, parser
+
+
+@parser("softnet_stat")
+class SoftNetStats(Parser):
+    """Parses ``/proc/net/softnet_stat`` file contains"""
+
+    def parse_content(self, contents):
+        cpu = 0
+        data = {}
+        for line in contents:
+            stats = line.split(' ')
+
+            data['cpu' + str(cpu)] = {'packet_process': stats[0],
+                                    'packet_drops': stats[1],
+                                    'time_squeeze': stats[2],
+                                    'cpu_collision': stats[8],
+                                    'received_rps': stats[9],
+                                    'flow_limit_count': stats[10]}
+            cpu = cpu + 1
+        self.data = data
+        self.data['cpu_instances'] = cpu
+
+    def get_cpu_instaces(self):
+        """
+        Get the number of cpu instances are handling network.
+
+        Arguments:
+            None: No input argument for the function.
+
+        Returns:
+            (int): The number of cpus handling network interrupts.
+        """
+        return self.data['cpu_instances']
+
+    def per_cpu_nstat(self, key):
```

Comment:
Better explain each `key` (line49-line54) in the document.

How about adding an interface that can be used like `nstat_of_cpu[0]` which returns a dict like (just a suggestion)
~~~
{'packet_process': ...,
 packet_drops':...,
 'time_squeeze': ...,
 'cpu_collision': ...,
'received_rps': ...,
 'flow_limit_count': ...}
~~~

Reply:
@xiangce - I think that'd be solved if this was simply an array by CPU number.

Reply:
Some times cpu instances becomes 2 even though ``nproc`` output it "4" , that depends on scheduling policies used. We will never look for the packet drops for let us say "cpu0" we can not predict such thing. we  will look for if there is over all packet drop (or any other non zero stats) on either of the processor, if there is drop then take corresponding action.